client/packages/system/src/Stock/api/hooks/useStockLine.ts

```diff
@@ -15,76 +16,185 @@ export interface DraftStockLine extends StockLineRowFragment {
   inventoryAdjustmentReason: InventoryAdjustmentReasonRowFragment | null;
 }
 
-export function useStockLine() {
-  const [draft, setDraft] = useState<DraftStockLine>({
-    __typename: 'StockLineNode',
-    id: '',
-    itemId: '',
-    onHold: false,
-    packSize: 0,
-    sellPricePerPack: 0,
-    costPricePerPack: 0,
-    totalNumberOfPacks: 0,
-    availableNumberOfPacks: 0,
-    storeId: '',
-    item: {
-      __typename: 'ItemNode',
-      code: '',
-      name: '',
-    },
-    inventoryAdjustmentReason: null,
-  });
+const defaultDraftStockLine = {
+  __typename: 'StockLineNode',
+  id: '',
+  itemId: '',
+  onHold: false,
+  packSize: 0,
+  sellPricePerPack: 0,
+  costPricePerPack: 0,
+  totalNumberOfPacks: 0,
+  availableNumberOfPacks: 0,
+  storeId: '',
+  item: {
+    __typename: 'ItemNode',
+    code: '',
+    name: '',
+  },
+  inventoryAdjustmentReason: null,
+};
+
+export function useStockLine(id?: string) {
+  const [patch, setPatch] = useState<Partial<DraftStockLine>>({});
+  const [isDirty, setIsDirty] = useState(false);
+  const { data, isLoading, error } = useGet(id ?? '');
+  const {
+    mutateAsync: createMutation,
+    isLoading: isCreating,
+    error: createError,
+  } = useCreate();
+  const {
+    mutateAsync: updateMutation,
+    isLoading: isUpdating,
+    error: updateError,
+  } = useUpdate(id ?? '');
+
+  const draft = (
+    data
+      ? { ...data?.nodes[0], ...patch }
+      : { ...defaultDraftStockLine, ...patch }
+  ) as DraftStockLine;
```

Comment:
Now draft is just a derived value -- from merging any changes values (in patch) into the original database data.

Reply:
```suggestion
  const draft = { ...defaultDraftStockLine, ...data?.nodes[0], ...patch };
```

Sorry missed this before - we shouldn't have `as` anywhere if we can help it (I think you're needing it here because data.nodes[0] could be undefined, and also because the queried stock line doesn't have an inventoryAdjustmentReasonId, which the draft requires...)